bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/editor/CEditorAssociationOverride.java

```diff
@@ -94,11 +98,21 @@ private boolean isEnabledFor(IEditorInput editorInput) {
 	}
 
 	private void deleteCodanMarkers(IResource resource) {
-		try {
-			resource.deleteMarkers(IProblemReporter.GENERIC_CODE_ANALYSIS_MARKER_TYPE, true, IResource.DEPTH_INFINITE);
-		} catch (CoreException e) {
-			Platform.getLog(CEditorAssociationOverride.class).error(e.getMessage(), e);
-		}
+		var wsJob = new WorkspaceJob("Remove codan markers") { //$NON-NLS-1$
+			@Override
+			public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
+				try {
+					resource.deleteMarkers(IProblemReporter.GENERIC_CODE_ANALYSIS_MARKER_TYPE, true,
+							IResource.DEPTH_INFINITE);
+				} catch (CoreException e) {
+					Platform.getLog(CEditorAssociationOverride.class).error(e.getMessage(), e);
```

Comment:
For `CoreException` it is better to use
```java
Platform.getLog(CEditorAssociationOverride.class).log(e.getStatus());
```

Reply:
Actually it is worse to use that version as you discard the stack trace. If deep in the code somewhere someone did this: `throw new CoreException(Status.error("brief message"))` then the log has no information on where the exception was raised.

Reply:
Yes, you are right, in the modern time it may be a case. 
I found less than 10 such constructs for CDT and dependencies, but they already exist. Even something as disappointing as `Status.error(null)`

Reply:
Even if someone has a really nice Status.error() with an embedded exception containing a stack trace, the stack trace will be from a different location. e.g. DSF's asynchronous code generates a Status object with stack trace in one thread, and throws the core exception in another thread. It is really useful to have both those stack traces.